Formula/m/mono.rb

```diff
@@ -16,27 +21,18 @@ class Mono < Formula
     sha256 x86_64_linux:   "e147b8ae7c32cda6c96a34bdd1de6f7d15c40af36b9ae68cb2196eb7e827e0a2"
   end
 
-  deprecate! date: "2023-10-24", because: "uses deprecated `openssl@1.1`"
-
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
+
+  depends_on arch: :x86_64
```

Comment:
Are we sure that mono will run on apple silicon macs in this way? Seems like they still have problem with running mono on arm. [mono/#21092](https://github.com/mono/mono/issues/21092).

Reply:
If we don't have a working mono, it will not be shipped on ARM. But this declares an explicit dependency on x86, which is false.

Reply:
We were originally able to build `mono` on ARM due to Rosetta on bare metal machines. At that time, I also had to disable `msbuild` since I could not get the current version to build.

Currently, `fsharp` won't build on our runners and no idea if support is planned. 

One last alternative would be to also disable `fsharp` from both Linux and ARM. I think that just leaves behind the `mono` runtime. The main problem here is `fsharp` needs to bootstrap off of .NET 3 (way too old for both Apple Silicon support and OpenSSL 3 support).

Reply:
> The main problem building `fsharp` is that it needs to bootstrap off of .NET 3 (way too old for both Apple Silicon support and OpenSSL 3 support).

any issue tracker for it? 

maybe let's disable fsharp? at least is better than nothing.

Reply:
> any issue tracker for it?
The development of mono is not very active anymore, and it seems that they are not interested in bumping the F# version for now.

> maybe let's disable fsharp? at least is better than nothing.
I'll try this.